web/js/modules/vector-styles/selectors.js

```diff
@@ -96,18 +95,51 @@ export function setRange(layerId, props, index, palettes, state) {
   };
 }
 
-export function setStyleFunction(def, vectorStyleId, vectorStyles, layer, state) {
+export function setStyleFunction(def, vectorStyleId, vectorStyles, layer, state, styleSelection = false) {
   const map = lodashGet(state, 'map.ui.selected');
   if (!map) return;
   const { proj } = state;
   const { selected } = state.vectorStyles;
   const { resolutions } = proj.selected;
   const layerId = def.id;
   const styleId = lodashGet(def, `vectorStyle.${proj.id}.id`) || vectorStyleId || lodashGet(def, 'vectorStyle.id') || layerId;
-  const glStyle = vectorStyles[styleId];
+  const customPalette = def.custom;
+
+  let glStyle = vectorStyles[styleId];
+  if (customPalette && Object.prototype.hasOwnProperty.call(state, 'palettes')) {
```

Comment:
[`hasOwn`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) is now recommended over `hasOwnProperty`. Also, is calling the method off of `Object.prototype` preferred to `state.hasOwn('palettes')`?

Reply:
It's recommended (& enforced in our lint rules) to call `hasOwnProperty` off `Object.prototype`. My understanding is that `hasOwnProperty `can be overwritten or otherwise corrupted/non-existent on a given object. So referring to the prototype is safer.

`Object.hasOwn()` is good to know - same result, less verbose, just as readable. Thanks.